pkg/server/server.go

```diff
@@ -115,6 +116,11 @@ type ExtraConfig struct {
 	//
 	// TemporaryRootShardKcpSharedInformerFactory bring data from the root shard
 	TemporaryRootShardKcpSharedInformerFactory kcpexternalversions.SharedInformerFactory
+
+	// ResolveRootKcpShardIdentities is to be called on server start until it succeeds. It injects the kcp
+	// resource identities for the root shard into the rest.Config used by the client.
+	// Only after it succeeds, the clients can wildcard-list/watch most kcp resources.
+	ResolveRootKcpShardIdentities func(ctx context.Context) error
```

Comment:
we have that at the top already

Reply:
this is for the root shard only, resolved in a different place. I think we need both.